### DISCUSSION TITLE
Feature extraction

### DISCUSSION THREAD

**Discussion by hailuu684** [martinsbruveris/tensorflow-image-models#64] [Q&A]
Hello everyone,
I would like to ask how can I extract the features for example, I extract the features from the resnet model and feed them to the ViT model. Does tfimm.create_preprocessing function do that?

**Comment by martinsbruveris** [comment]
Apologies for the (very) late reply. All `tfimm` models return intermediate features if called with `return_features=True`. E.g.,
```
model = tfimm.create_model("resnet18")
img = tf.zeros((1, 224, 224, 3))
logits, features = model(img, return_features=True)
```
Now `features` will be an `OrderedDict` containing the intermediate features.
